mhkit/acoustics/graphics.py

```diff
@@ -0,0 +1,102 @@
+"""
+This submodule includes the main passive acoustics plotting functions. All
+functions allow passthrough of matplotlib functionality and commands
+to make them fully customizable.
+"""
+
+import matplotlib.pyplot as plt
+from .analysis import _fmax_warning
+
+
+def plot_spectogram(spsdl, fmin=10, fmax=100000, fig=None, ax=None, **kwargs):
+    """
+    Plots the spectogram of the sound pressure spectral density level.
+
+    Parameters
+    ----------
+    spsdl: xarray DataArray (time, freq)
+        Mean square sound pressure spectral density level in dB rel 1 uPa^2/Hz
+    fmin: int
+        Lower frequency band limit (lower limit of the hydrophone). Default: 10 Hz
+    fmax: int
+        Upper frequency band limit (Nyquist frequency). Default: 100000 Hz
+    fig: matplotlib.pyplot.figure
+        Figure handle to plot on
+    ax: matplotlib.pyplot.axis
+        Figure axis containing plot objects
+    kwargs: dict
+        Dictionary of matplotlib function keyword arguments
+
+    Returns
+    -------
+    fig: matplotlib.pyplot.figure
+        Figure handle of plot
+    ax: matplotlib.pyplot.axis
+        Figure plot axis
+    """
+
+    # Set dimension names
+    time = spsdl.dims[0]
+    freq = spsdl.dims[-1]
```

Comment:
Well no, because you can have a "time_bins" or "freq_bins" coordinates, which are always first and second, respectively, using this codebase.

If you're smart enough to know how to a read a hydrophone file yourself you probably also have plotting functionality.